apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedisConnectionMethodInterceptor.java

```diff
@@ -0,0 +1,114 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+
+package org.apache.skywalking.apm.plugin.redisson.v3;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+import org.redisson.client.RedisClient;
+import org.redisson.client.RedisConnection;
+import org.redisson.client.protocol.CommandData;
+import org.redisson.client.protocol.CommandsData;
+
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+
+/**
+ * @author zhaoyuguang
+ */
+public class RedisConnectionMethodInterceptor implements InstanceMethodsAroundInterceptor, InstanceConstructorInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        String peer = (String) objInst.getSkyWalkingDynamicField();
+
+        RedisConnection connection = (RedisConnection) objInst;
+        Channel channel = connection.getChannel();
+        InetSocketAddress remoteAddress = (InetSocketAddress) channel.remoteAddress();
+        String dbInstance = remoteAddress.getAddress().getHostAddress() + ":" + remoteAddress.getPort();
+
+        StringBuilder dbStatement = new StringBuilder();
+        String operationName = "Redisson/";
+
+        if (allArguments[0] instanceof CommandsData) {
+            operationName = operationName + "BATCH_EXECUTE";
+            CommandsData commands = (CommandsData) allArguments[0];
+            for (CommandData commandData : commands.getCommands()) {
+                addCommandData(dbStatement, commandData);
+                dbStatement.append(";");
+            }
+        }
+        if (allArguments[0] instanceof CommandData) {
```

Comment:
This should be `else if` to avoid one unnecessary `instanceof`.

Reply:
Yeah, that's better